### PR TITLE
[DBX-89576] add new 0718 upload flipping logic

### PR DESCRIPTION
**Pull request by SamStuckey**
## Summary

- *This work is behind a feature toggle (flipper): NO*
- Add a flipper to commit users to a new version of the 0781 form flow
- [Supports FE work in this PR](https://github.com/department-of-veterans-affairs/vets-website/pull/32237)

## Related issue(s)

- [Ticket to add flipper](https://github.com/department-of-veterans-affairs/va.gov-team/issues/92013)
- [Ticket to limit to new users / commit users to a flow](https://github.com/department-of-veterans-affairs/va.gov-team/issues/92199)

**NOTE: this change accomplishes both goals by leveraging the SaveInProgressForm logic on the FE**

## Testing done

- [x] *New code is covered by unit tests* (n/a)
- local env click through testing by modifying the value in `config/feature.yml` and confirming setting and saving of value
- this is code that supports upcoming work and will not be immediately user facing

## Screenshots
n/A

## What areas of the site does it impact?
Nothing yet, but this will support our sync with 0781 paper version in the 526 form flow

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [x]  Documentation has been updated (see tickets)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x]  Feature/bug has a monitor built into Datadog (n/A)
- [x]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature (n/a)

### CODE REVIEW

File: config/features.yml [hunk 1]
```diff
@@ -466,6 +466,10 @@ features:
     actor_type: user
     description: enable to test the backup submission path. DO NOT TOGGLE THIS FLAG UNLESS YOU ARE A MEMBER OF DISABILITY BENEFITS EXPERIENCE TEAM.
     enable_in_development: true
+  disability_compensation_sync_modern_0781_flow:
+    actor_type: user
+    description: enables a new form flow to for 0781 and 0781a in the 526 submission workflow
+    enable_in_development: false
```
Comment (freeheeling) [inline config/features.yml:472]: Are you waiting until later to enable this feature in development?

File: config/features.yml [hunk 2]
```diff
@@ -466,6 +466,10 @@ features:
     actor_type: user
     description: enable to test the backup submission path. DO NOT TOGGLE THIS FLAG UNLESS YOU ARE A MEMBER OF DISABILITY BENEFITS EXPERIENCE TEAM.
     enable_in_development: true
+  disability_compensation_sync_modern_0781_flow:
+    actor_type: user
+    description: enables a new form flow to for 0781 and 0781a in the 526 submission workflow
```
Comment (freeheeling) [inline config/features.yml:471]: typo: 'flow **to** for 0781'

File: config/features.yml (hunk 1 above)
Comment (SamStuckey) [reply config/features.yml:472]: no reason to have it on ¯\_(ツ)_/¯ i figured the value here is arbitrary but if leaving it on is better practice i'll update

File: config/features.yml (hunk 2 above)
Comment (SamStuckey) [reply config/features.yml]: updated

File: config/features.yml (hunk 1 above)
Comment (SamStuckey) [reply config/features.yml]: updated

**Review by tommasina-va — APPROVED** [review]

**Review by LindseySaari — APPROVED** [review]

**Pull request merged**